Automattic/jetpack
# VideoPress: Upload video from local library

⚠️ There's some known problems with re-uploading the same video

Closes #26501 

#### Changes proposed in this Pull Request:

* Extract upload from library logic from blocks.
* Add support for `disabled` and `action button` related props in `VideoRow`.
* Improve types for `LocalVideo` related components.

#### Other information:


#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






_Plugin_

* Start `VideoPress`.
* Goes to `Local Videos` section and click to upload.
* It should upload the video and mark it in local as uploaded.
* Now upload a regular video, directly to VideoPress.
* It should work too.

_Block_

* Insert `VideoPress` block.
* Select video from library.
* It should upload normally to VideoPress.

#### Demo


https://user-images.githubusercontent.com/1663717/197053129-faa8f47b-705f-42f8-b631-4b44e26976c0.mp4
